modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -996,6 +996,34 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
                              CV_OUT std::vector<int>& indices,
                              const float eta = 1.f, const int top_k = 0);
 
+     class CV_EXPORTS_W_SIMPLE Model : public Net
+     {
+     public:
+         Model(const std::string& model, const std::string& config = "", int width = -1,
+               int height = -1, Scalar mean = Scalar(), float scale = 1.0,
+               bool swapRB = true, bool crop = false);
+
+         void setInputWidth(int width);
```

Comment:
Constructor has too many parameter.
This would cause the same problems, like `blobFromImage()`.

> void setInputWidth

Return self reference instead of `void`:
> Model& setInputWidth()

It would allow to build configuration chains with well specified parameters names:

```
m = Model(loaded_network)
    .setParameter1(param1)
    .setParameter2(param2)
;
```

Reply:
@l-bat, Please remove preprocessing params from constructors as @alalek pointed.